### PR TITLE
(RK-142) Add machine readable format options to "deploy display"

### PR DESCRIPTION
**Pull request by scotje**
This may merit some test coverage at this point...

### CODE REVIEW

**Comment by scotje** [comment]
Thinking about just removing the "legacy" output format entirely and making "yaml" the default. Only question is whether anyone has integrated against the legacy pseudo-yaml output already and if it's worth preserving for that case.

File: lib/r10k/cli/deploy.rb [hunk 1]
```diff
@@ -88,6 +88,7 @@ def self.command
 
           flag :p, :puppetfile, 'Display Puppetfile modules'
           flag nil, :detail, 'Display detailed information'
+          optional nil, :format, 'Display output in a specific format. Valid values: json, yaml', :argument => :required
```
Comment (adrienthebo) [inline lib/r10k/cli/deploy.rb]: This can be `required` instead of `optional`, which drops the need to have `:argument => :required`

Comment (scotje) [inline lib/r10k/cli/deploy.rb]: Ah ok, yeah I found the documentation around that part of Cri a little thin....

**Comment by adrienthebo** [comment]
I think we should drop the legacy format; it's really crufty and the YAML/JSON outputters give people a real, parseable format.

**Comment by scotje** [comment]
Pushed a new commit, if that looks good I can squash it so we can merge.

**Comment by adrienthebo** [comment]
:+1:, squash it and I'll merge it.

**Comment by scotje** [comment]
squashed

**Pull request merged**